opal/mca/common/ucx/common_ucx_wpool.c

```diff
@@ -308,6 +309,10 @@ opal_common_ucx_wpool_progress(opal_common_ucx_wpool_t *wpool)
             }
             opal_mutex_unlock(&winfo->mutex);
         }
+        if (active_workers == 0 && opal_list_get_size(&wpool->idle_workers)) {
```

Comment:
Actually, I think we need to progress it in any case.
It might happen that it's in the "idle" list even if we have other workers non-idling.

Reply:
We are currently doing a significant refactoring of this code BTW. 
But this is a good hint for us. So thank you very much.
For now, I think, we can just always progress this one. But with locking precautions.

Reply:
The `wpool->mutex` is held at this point, that should be sufficient, right?

Reply:
@devreal we need to progress the default worker.